testing/tests/DevExpress.ui.widgets.editors/textEditorParts/mask.tests.js

```diff
@@ -418,6 +419,48 @@ QUnit.module('typing', moduleConfig, () => {
         assert.equal(handler.callCount, 2, '\'change\' event is not fired after focus out without value change');
         clock.restore();
     });
+
+    QUnit.test('TextEditor with mask option should work correctly with autofill (T869537)', function(assert) {
+        if(!browser.webkit && !(browser.msie && browser.version > 11)) {
+            assert.expect(0);
+            return;
+        }
+
+        const clock = sinon.useFakeTimers();
+        const testText = '555555';
+        const $textEditor = $('#texteditor').dxTextEditor({
+            mask: '+1 (X00) 000',
+            maskRules: { X: /[02-9]/ },
+            name: 'phonenumber',
+            pattern: '[0-9]*',
+            mode: 'tel',
+            useMaskedValue: true
+        });
+        const $input = $textEditor.find('.dx-texteditor-input');
+        const textEditor = $textEditor.dxTextEditor('instance');
+        const keyboard = keyboardMock($input, true);
+
+        $input.val(testText);
+        let inputMatchesStub;
+        if(browser.webkit) {
+            inputMatchesStub = sinon.stub($input.get(0), 'matches', () => true);
+        } else {
+            $input.addClass('edge-autofilled');
+        }
+
+        if(devices.real().android) {
+            keyboard.beforeInput(testText, 'insertText');
+        }
+
+        $input.trigger($.Event('input', { originalEvent: $.Event('input') }));
```

Comment:
keyboard.type(testText)?

Reply:
no, to emulate autofill we need to call input event only

Reply:
keyboard
   .beforeInput(testText)
   .input(testText)

?

Reply:
we need to use input without text to simulate autofill behavior

Reply:
>we need to use input without text to simulate autofill behavior

keyboard
   .beforeInput(testText)
   .input()

?